src/error.rs

```diff
@@ -24,6 +27,11 @@ pub enum EvalError<'tcx> {
     Unimplemented(String),
     DerefFunctionPointer,
     ExecuteMemory,
+    ArrayIndexOutOfBounds(Span, u64, u64),
+    Math(Span, ConstMathErr),
+    InvalidBitShiftRhs(PrimVal),
```

Comment:
This is not actually a thing, the semantics of shifting in Rust is that the RHS gets masked `(1 << BITS) - 1` and interpreted as unsigned.